### PR TITLE
Added TTL to state store binding for DynamoDB

### PR DESCRIPTION
**Pull request by wimvdc**
# Description

Adding ttlInSeconds support to DynamoDB as state store.
Please see PR #1493 for discussion details

## Issue reference

#1489

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: https://github.com/dapr/docs/pull/2203

### CODE REVIEW

**Comment by wimvdc** [comment]
See following PR for the documentation update.
https://github.com/dapr/docs/pull/2203

**Comment by yaron2** [comment]
@berndverst can you please re-review this PR?

**Comment by yaron2** [comment]
@wimvdc thanks for this PR, please see the failing lint (file needs to be `gofumpt`-ed).

**Comment by wimvdc** [comment]
> @wimvdc thanks for this PR, please see the failing lint (file needs to be `gofumpt`-ed).

Thank you for your help and time on the Dapr projects.
I gofumpt-ed both files. 
Might be useful to add this step to some documentation/script (?)

**Comment by daixiang0** [comment]
> > @wimvdc thanks for this PR, please see the failing lint (file needs to be `gofumpt`-ed).
> 
> Thank you for your help and time on the Dapr projects. I gofumpt-ed both files. Might be useful to add this step to some documentation/script (?)

I think lint is common in all project, Go project usually use `golangci-lint` which include `gofumpt`. You can run `make lint` to detect lint issues, and you can get shell commands from CI logs in any project. Hope it helps.

File: state/aws/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -100,6 +105,17 @@ func (d *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
 		return nil, err
 	}
 
+	var ttl int64
+	if result.Item[d.ttlAttributeName] != nil {
```
Comment (berndverst) [inline state/aws/dynamodb/dynamodb.go:109]: Careful, `meta.TTLAttribute` should not be required. So this means the value will be an empty string here. You cannot index into your item with an empty string.

I recommend doing the following:

```suggestion
        if d.ttlAttributeName != "" {
	if val, ok := result.Item[d.ttlAttributeName]; ok {
	   if err = dynamodbattribute.Unmarshal(val, &ttl); err != nil {
```

File: state/aws/dynamodb/dynamodb_test.go [hunk 2]
```diff
@@ -32,6 +35,12 @@ type mockedDynamoDB struct {
 	dynamodbiface.DynamoDBAPI
 }
 
+type DynmoDBItem struct {
```
Comment (berndverst) [inline state/aws/dynamodb/dynamodb_test.go:38]:
```suggestion
type DynamoDBItem struct {
```

**Review by berndverst — CHANGES_REQUESTED** [review]
Looks great. Just one important change to be made here on the Get request

 And make sure `dynamoDB` is spelled correctly / consistently throughout :)

**Review by berndverst — DISMISSED** [review]
Looks great. Just one important change to be made on the Get request.

Also make sure to spell `dynamodb` variables / structs etc consistently :)

File: state/aws/dynamodb/dynamodb_test.go (hunk 2 above)
Comment (wimvdc) [reply state/aws/dynamodb/dynamodb_test.go]: Fixed

**Pull request merged**